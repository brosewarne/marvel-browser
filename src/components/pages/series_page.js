import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';

import { ItemImage } from '../items/item_image';

import './pages.css';

/**
 * Render the SeriesPage, showing a profile of the selected series
 * @param {Object} props - The set of props for the SeriesPage - passed from the route
 * @returns {Node} - The rendered SeriesPage
 * @constructor
 */
export const _SeriesPage = (props) => {
    const {
        title, description, startYear, endYear, type, stories, comics, characters
    } = props.item;

    return (
        <div>
            <content className="pageContainer">
                <Paper elevation={12} className="itemProfile">
                    <h1 className="itemProfileTitle">{title}</h1>
                    <ItemImage item={props.item} imgSize="portrait_uncanny" />
                    <p>{description || 'No series description available'}</p>
                    <p>Type: {type}</p>
                    <p>Start Year: {startYear}</p>
                    <p>End Year: {endYear}</p>
                    <p>{`${characters.available} Characters`}</p>
                    <p>{`${comics.available} Comics`}</p>
                    <p>{`${stories.available} Stories`}</p>
                </Paper>
            </content>
        </div>
    );
};

_SeriesPage.propTypes = {
    item: PropTypes.object.isRequired
};

/**
 * If item is not supplied from props, get it from the app state
 * @param {Object} state - The current app state
 * @param {Object} props - The incoming set of props
 * @returns {Object} - Empty id item is already in props, otherwise, the mapped item from app state
 */
const mapStateToProps = (state, props) => {
    return props.item ? {} : {
        item: state.routing.location.state.data
    };
};

export const SeriesPage = connect(mapStateToProps)(_SeriesPage);
