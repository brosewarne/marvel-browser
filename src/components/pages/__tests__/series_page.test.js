import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { _SeriesPage } from '../series_page';

configure({ adapter: new Adapter() });

describe('<_SeriesPage />', () => {
    const mockSeries = {
        title: 'Series Title',
        description: 'Series Description',
        startYear: '1990',
        endYear: '2000',
        type: 'Series Type',
        thumbnail: {
            path: '/a/href/to/an/image',
            extension: 'png'
        },
        comics: {
            available: 10
        },
        characters: {
            available: 2
        },
        stories: {
            available: 5
        },
        urls: [
            {
                url: '/aURL',
                type: 'wiki'
            }
        ]
    };
    it('renders correctly', () => {
        const wrapper = shallow(
            <_SeriesPage item={mockSeries} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
