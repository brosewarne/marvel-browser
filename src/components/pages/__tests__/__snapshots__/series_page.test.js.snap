// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<_SeriesPage /> renders correctly 1`] = `
<div>
  <content
    className="pageContainer"
  >
    <WithStyles(Paper)
      className="itemProfile"
      elevation={12}
    >
      <h1
        className="itemProfileTitle"
      >
        Series Title
      </h1>
      <Component
        imgSize="portrait_uncanny"
        item={
          Object {
            "characters": Object {
              "available": 2,
            },
            "comics": Object {
              "available": 10,
            },
            "description": "Series Description",
            "endYear": "2000",
            "startYear": "1990",
            "stories": Object {
              "available": 5,
            },
            "thumbnail": Object {
              "extension": "png",
              "path": "/a/href/to/an/image",
            },
            "title": "Series Title",
            "type": "Series Type",
            "urls": Array [
              Object {
                "type": "wiki",
                "url": "/aURL",
              },
            ],
          }
        }
      />
      <p>
        Series Description
      </p>
      <p>
        Type: 
        Series Type
      </p>
      <p>
        Start Year: 
        1990
      </p>
      <p>
        End Year: 
        2000
      </p>
      <p>
        2 Characters
      </p>
      <p>
        10 Comics
      </p>
      <p>
        5 Stories
      </p>
    </WithStyles(Paper)>
  </content>
</div>
`;
